### PR TITLE
Fix bash completions on Windows when WSL is installed

### LINKED ISSUE #2140: git completion on windows broken
Git completions on windows has stopped working:
```
snail@sea ~\Documents\xonsh master $ $BASH_COMPLETIONS
['C:\\Program Files\\Git\\usr\\share\\bash-completion\\bash_completion', 
'C:\\Program Files\\Git\\mingw64\\share\\git\\completion\\git-completion.bash']
```

I guess it could be some of the recent fixes to bash completion scripts. Do we require bash completion 2.0? 

On windows we use the git bash completions that come bundled with 'Git for Windows'. I don't think that 2.0

### PR DESCRIPTION
**Pull request by melund**
This fixes the problem I had with bash completion on windows because 'Windows Subsystem for Linux' was installed. 

Fixes #2140

### CODE REVIEW

File: xonsh/platform.py [hunk 1]
```diff
@@ -6,14 +6,14 @@
...
 import pathlib
+import builtins
 import platform
 import functools
 import subprocess
 import importlib.util
 
 from xonsh.lazyasd import LazyBool, lazyobject, lazybool
 
-
```
Comment (mitnk) [inline xonsh/platform.py]: I guess two black lines are for pep8, or we will break tools like `flake8` etc.

Comment (scopatz) [inline xonsh/platform.py]: Yeah that seems right to me.

**Comment by scopatz** [comment]
I don't think we need to wait for Travis today. Thanks for putting this in @melund! LGTM

File: xonsh/platform.py [hunk 2]
```diff
@@ -285,16 +285,21 @@ def windows_bash_command():
...
-        subprocess.check_call([wbc, '--version'],
-                              stdout=subprocess.PIPE,
-                              stderr=subprocess.PIPE)
-    except (FileNotFoundError, subprocess.CalledProcessError):
-        gfwp = git_for_windows_path()
-        if gfwp:
-            bashcmd = os.path.join(gfwp, 'bin\\bash.exe')
-            if os.path.isfile(bashcmd):
-                wbc = bashcmd
+    bash_on_path = builtins.__xonsh_commands_cache__.lazy_locate_binary('bash')
```
Comment (scopatz) [inline xonsh/platform.py]: Maybe we want to add `ignore_alias=True` here since we want the real path

File: xonsh/platform.py (hunk 1 above)
Comment (melund) [inline xonsh/platform.py]: Thanks. I wondered what it was complaining about.

File: xonsh/platform.py (hunk 2 above)
Comment (melund) [inline xonsh/platform.py]: Good idea.

**Comment by melund** [comment]
I away from my computer until tomorrow, but I tired to make the commit on my phone....

**Comment by scopatz** [comment]
Looks like you got it. Thanks!

**Pull request merged**